### PR TITLE
feat(metric-stats): Emit accepted volume

### PR DESCRIPTION
**Pull request by Dav1dde**
Emits positive metric stats for the custom metrics namespace after submission to Kafka.

Only implements the happy path for metric stats and only volume.

### CODE REVIEW

File: relay-dynamic-config/src/feature.rs [hunk 1]
```diff
@@ -45,6 +45,11 @@ pub enum Feature {
     /// Serialized as `organizations:custom-metrics`.
     #[serde(rename = "organizations:custom-metrics")]
     CustomMetrics,
+    /// Enables metric stats.
+    ///
+    /// Serialized as `organizations:metric-stats`.
+    #[serde(rename = "organizations:metric-stats")]
+    MetricStats,
```
Comment (Dav1dde) [inline relay-dynamic-config/src/feature.rs]: Considering changing this to a global config option

File: relay-metrics/src/view.rs [hunk 2]
```diff
@@ -376,6 +377,11 @@ impl<'a> BucketView<'a> {
         &self.inner.name
     }
 
+    /// Returns the name of the bucket.
```
Comment (jjbayer) [inline relay-metrics/src/view.rs:380]: nit: doc comment does not really distinguish it from the method above, maybe add something like "Caller holds shared ownership of the string"?

File: relay-server/src/services/metric_stats.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+impl Service for MetricStatsService {
+    type Interface = TrackMetric;
+
+    fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            while let Some(message) = rx.recv().await {
+                self.handle_track_metric(message);
+            }
+            dbg!("END");
```
Comment (jjbayer) [inline relay-server/src/services/metric_stats.rs]:
```suggestion
            relay_log::info!("MetricStatsService has stopped.");
```

**Review by jjbayer — COMMENTED** [review]
Looks good to me!

As discussed:

* Slight preference for a global config org rollout rate.
* Preference for a thread-less "service" / facade for `MetricStatsService`.

File: relay-server/src/metric_stats.rs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+/// Tracks stats about metrics.
+///
+/// Metric stats are similar to outcomes for envelopes, they record
+/// the final "fate" of a metric and its properties.
+///
+/// Unlike outcomes metric stats are tracked on a per MRI basis
+/// and contain additional metadata, like the cardinality of a metric.
+#[derive(Clone, Debug)]
+pub struct MetricStats {
+    config: Arc<Config>,
```
Comment (jjbayer) [inline relay-server/src/metric_stats.rs:30]: nit: Is `config` only used to check `processing_enabled`? If so, we could give MetricStats a noop variant instead, something like

```rust
enum MetricStats {
    Disabled,
    Enabled {
        global_config: GlobalConfigHandle,
        aggregator: Addr<Aggregator>,
    }
}
```

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/metric_stats.rs (hunk 4 above)
Comment (Dav1dde) [reply relay-server/src/metric_stats.rs:30]: That makes dealing with `self` awkward, since I have to pattern match it in every method.

**Pull request merged**